Source/Parallelization/WarpXComm.cpp

```diff
@@ -181,7 +181,9 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
             // ParallelCopy from coarse level
             for (int i = 0; i < 3; ++i) {
                 IntVect ng = Btmp[i]->nGrowVect();
-                Btmp[i]->ParallelCopy(*Bfield_aux[lev-1][i], 0, 0, 1, ng, ng, cperiod);
+                // Guard cells may not be up to date beyond ng_FieldGather
+                const auto ng_src = guard_cells.ng_FieldGather;
```

Comment:
The code you wrote is perfect, but what do you think about writing the type of `ng_src` explicitly (I guess with or without reference `&` does not make much difference...), here as well as below? I personally find it easier when I can learn the type of a variable immediately from the code:
```suggestion
                const amrex::IntVect& ng_src = guard_cells.ng_FieldGather;
```

Reply:
Sounds good, I've added the explicit type everytime I declared `ng_src`.